security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -116,12 +118,16 @@ func (p Plugin) ExchangeToken(ctx context.Context, trustDomain, k8sSAjwt string)
 	return respData.AccessToken, time.Now().Add(time.Second * time.Duration(respData.ExpiresIn)), resp.StatusCode, nil
 }
 
-func constructAudience(trustDomain string) string {
-	if GKEClusterURL == "" {
-		return trustDomain
+func constructAudience(credFetcher security.CredFetcher, trustDomain string) string {
+	provider := credFetcher.GetIdentityProvider()
```

Comment:
This code doesn't make any sense.
@howardjohn  do you understand it ?  This is google-specific code - will it even work with the FakeGKECluster ? And why would the provider be empty - and not return the actual value ? 

Since this functionality is google-specific - and hopefully will be replaced by the core gRPC libraries - I'll not block on this, but it doesn't look clean.

Reply:
At least add a comment on what changed - before this PR we were sending the GKEClusterURL.

Is there any case where we won't (except for mock testing - which should not be part of the code, but tests setting a custom GKEClusterURL) ? Are there any server-side changes to the protocol ?